python/foglamp/services/core/api/plugins/install.py

```diff
@@ -0,0 +1,179 @@
+            if compressed not in ['true', 'false', True, False]:
+                raise ValueError('Only "true", "false", true, false are allowed for value of compressed.')
+        is_compressed = ((isinstance(compressed, str) and compressed.lower() in ['true']) or (
+            (isinstance(compressed, bool) and compressed is True)))
+
+        # All stuff goes into _PATH
+        if not os.path.exists(_PATH):
+            os.makedirs(_PATH)
+
+        result = await download([url])
+        # _LOGGER.info("Result {} ".format(result))
+        file_name = result[0].split('Successfully downloaded ')[-1]
+        # _LOGGER.info("file_name {} {}".format(file_name, checksum))
+
+        # TODO: only validate with MD5sum. Do we need to validate with SHA1, SHA256, SHA512?
+        # if yes then we need to add checksum type attribute in request
+        if validate_checksum(checksum, file_name) is False:
+            raise ValueError("Checksum is failed.")
+
+        # TODO: tar this downloaded file under _PATH and install plugin FOGLAMP_ROOT
+        _LOGGER.info("Found {} format with compressed {}".format(file_format, is_compressed))
+        _files = extract_file(file_name, is_compressed)
+        _LOGGER.info("Files {} {}".format(_files, type(_files)))
+        if file_format == 'tar':
+            copy_file_install_requirement(_files, _type)
+        else:
+            install_debian(_files[0])
+    except (TypeError, ValueError) as ex:
+        raise web.HTTPBadRequest(reason=str(ex))
+    except Exception as ex:
+        raise web.HTTPException(reason=str(ex))
+    else:
+        return web.json_response({"message": "{} and installed".format(result)})
+
+
+async def get_url(url: str, session: aiohttp.ClientSession) -> str:
+    file_name = str(url.split("/")[-1])
+    async with async_timeout.timeout(TIME_OUT):
+        async with session.get(url) as response:
+            with open(_PATH + file_name, 'wb') as fd:
+                async for data in response.content.iter_chunked(CHUNK_SIZE):
+                    fd.write(data)
+    return 'Successfully downloaded ' + file_name
+
+
+async def download(urls: list) -> asyncio.gather:
+    async with aiohttp.ClientSession() as session:
+        tasks = [get_url(url, session) for url in urls]
+        return await asyncio.gather(*tasks)
+
+
+def validate_checksum(checksum: str, file_name: str) -> bool:
+    original = hashlib.md5(open(_PATH + file_name, 'rb').read()).hexdigest()
+    return True if original == checksum else False
+
+
+def extract_file(file_name: str, is_compressed: bool) -> list:
+    mode = "r:gz" if is_compressed else "r"
+    tar = tarfile.open(_PATH + file_name, mode)
+    _LOGGER.info("Extracted to {}".format(_PATH))
+    tar.extractall(_PATH)
+    _LOGGER.info("Extraction Done!!")
+    # _LOGGER.info(tar.getnames())
+    return tar.getnames()
+
+
+def install_debian(file_name: str):
+    # FIXME: Not working seems like we need to manipulate in /etc/sudoers.d/foglamp file
```

Comment:
Possibly, or recreate a shell script that will install packages and make it set user id on execution chmod 455 <script>

The same trick can be done to install a tar archive if the FogLAMP is installed via sudo make install. Th make install rule will need to set the ownership and permissions on that script as part of the make install

Reply:
yeah will try the solution which is provided in https://github.com/foglamp/FogLAMP/pull/1494